src/Sarif/Core/Stack.cs

```diff
@@ -0,0 +1,166 @@
+                current = exceptions.Dequeue();
+
+                if (current.InnerException != null)
+                {
+                    exceptions.Enqueue(current.InnerException);
+                }
+
+                stack = Create(current.StackTrace);
+
+                stack.Message = current.Message;
+
+                stacks.Add(stack);
+            }
+
+            return stacks;
+        }
+
+        public static Stack Create(string stackTrace)
+        {
+            Stack stack = new Stack();
+            
+            if (string.IsNullOrEmpty(stackTrace))
+            {
+                return stack;
+            }
+
+            stack.Frames = new List<StackFrame>();
+
+            foreach (string line in stackTrace.Split('\n'))
+            {
+                // at Type.Method() in File.cs : line X\r
+                string current = line;
+
+                StackFrame stackFrame;
+                stackFrame = new StackFrame();
+
+                // at Type.Method() in File.cs: line X
+                current = line.Trim('\r');
+
+                // Type.Method() in File.cs: line X
+                current = current.Substring(AT.Length);
+                stackFrame.FullyQualifiedLogicalName = current.Split(')')[0] + ")";
+
+                // in File.cs: line X
+                current = current.Substring(stackFrame.FullyQualifiedLogicalName.Length);
+
+                if (current.Length > 0)
+                {
+                    int lineNumber;
+                    string fileName;
+
+                    // File.cs: line X
+                    current = current.Substring(IN.Length);
+
+                    // :line X
+                    fileName = current.Split(' ')[0];
+                    fileName = fileName.Substring(0, fileName.Length - LINE.Length);
+
+                    // X
+                    current = current.Substring(fileName.Length + LINE.Length);
+                    lineNumber = int.Parse(current);
+
+                    stackFrame.Location = new PhysicalLocation()
+                    {
+                        Uri = new Uri(fileName),
+                        Region = new Region()
+                        {
+                           StartLine = lineNumber
+                        }
+                    };
+                }
+                stack.Frames.Add(stackFrame);
+            }
+
+            return stack;
+        }
+
+        public override string ToString()
+        {
+            return ToString(StackFormat.Default);
+        }
+
+        public string ToString(StackFormat stackFormat)
+        {           
+            if (this.Frames == null) { return null; }
+
+            StringBuilder sb = new StringBuilder(255);
+
+            for (int iFrameIndex = 0; iFrameIndex < this.Frames.Count; iFrameIndex++)
+            {
+                StackFrame sf = this.Frames[iFrameIndex];
+
+                if (iFrameIndex > 0) { sb.AppendLine(); }
+
+                sb.Append(sf.ToString());
```

Comment:
I put the StrackTrace formatting of StackFrames into the StackFrame ToString() override. .NET doesn't do this, though, it formats frames specifically for stacks and then provides a more comprehensive ToString() for StackFrame (that is really only suitable for debugging due to failure to handle missing data conditions in a readable way).